test/integration/integration_test.py

```diff
@@ -0,0 +1,44 @@
+# Copyright (c) 2023 Robert Bosch GmbH
+#
+# This program and the accompanying materials are made available under the
+# terms of the Apache License, Version 2.0 which is available at
+# https://www.apache.org/licenses/LICENSE-2.0.
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+#
+# SPDX-License-Identifier: Apache-2.0
+
+import subprocess, re
+
+
+def running_script_with_log(command, regex):
+    successfully_running = False
+    built_regex = re.compile(regex)
+    proc = subprocess.Popen(command.split(" "), stdout=subprocess.PIPE, bufsize=1, universal_newlines=True)
+    for line in iter(proc.stdout.readline, ""):
+        print(line)
+        if re.match(built_regex, line):
+            successfully_running = True
+            break
+        if proc.poll() is not None:
+            break
+    return successfully_running
+
+
+def ensure_dapr(command):
```

Comment:
Don't introduce a function just to hide a single subprocess invocation., you can do this directly in your main test function.